src/mlpack/core/data/dictionary_encoding_impl.hpp

```diff
@@ -0,0 +1,167 @@
+/**
+ * @file dictionary_encoding_impl.hpp
+ * @author Jeffin Sam
+ *
+ * Implementation of dictionary encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_DICT_ENCODING_IMPL_HPP
+#define MLPACK_CORE_DATA_DICT_ENCODING_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "dictionary_encoding.hpp"
+
+namespace mlpack {
+namespace data {
+
+void DicitonaryEncoding::Reset()
+{
+  mappings.clear();
+  stringq.clear();
+}
+
+DicitonaryEncoding::DicitonaryEncoding(const DicitonaryEncoding &old_obj)
+{
+  this->mappings.clear();
+  this->stringq.clear();
+  for (auto it = old_obj.mappings.begin(); it != old_obj.mappings.end(); it++)
+  {
+    this->stringq.push_back(std::string(it->first));
+    this->mappings[stringq.back()] = it->second;
+  }
+}
+
+void DicitonaryEncoding::operator= (const
+    DicitonaryEncoding &old_obj)
+{
+  if (this != &old_obj)
+  {
+    this->mappings.clear();
+    this->stringq.clear();
+    for (auto it = old_obj.mappings.begin(); it != old_obj.mappings.end(); it++)
+    {
+      this->stringq.push_back(std::string(it->first));
+      this->mappings[stringq.back()] = it->second;
+    }
+  }
+  else
+  {
+    this->mappings = old_obj.mappings;
```

Comment:
I think in this case there is nothing to do:)

Reply:
Made changes.